pkg/evaluator/value.go

```diff
@@ -334,22 +334,25 @@ func isBreak(val value) bool {
 func normalizeSliceIndices(start, end value, length int) (int, int, error) {
 	startIdx := 0
 	var err error
-	if start != nil {
```

Comment:
from https://go.dev/ref/spec#Types on type assertions:
> asserts that x is not nil and that the value stored in x is of type T.

therefore the nil checks on start and end are redundant

Reply:
The nil checks are there to handle when the value is not supplied. We still want to call `normalizeIndex` which does the type assertion as it handles the case of the wrong type for the index, as opposed to a missing index. A single type assertion cannot tell those cases apart.

This evy code produces a result with the code in this PR but should be an error:
```
a := [1 2 3]
b := "x"
print a[b:b]
```

(although the current code generates an `InternalError` for this which is not correct. It seems perhaps what's needed here is for the parser to type check this properly and then perhaps the type assertion approach you're taking here would work. But it still does remove that internal error check (which really should not happen, but that's the case for all the internal errors).